sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -210,6 +211,9 @@ namespace Azure { namespace Core { namespace Http {
     void StartTry();
 
   public:
+    bool DoNotRetry = false;
+    std::chrono::milliseconds ConnectionTimeout = std::chrono::milliseconds(0); // 0 means default
```

Comment:
Two nits: 
* I would rename `DoNotRetry` to `DisableRetries`.
* You can use `std:::chrono::milliseconds ConnectionTimeout = {}` rather than specifying 0 milliseconds.